### ISSUE TITLE
unrecognized cop or department `inherit_gem`

### ISSUE THREAD

**Issue by faraaz-deepsource** [rubocop/rubocop#11146]
I have the following `.rubocop.yml` defined:
```yaml
inherit_gem:
  ted_rubocop_rules: rubocop.yml
```
and then I manually create a config using `RuboCop::Config#new` while passing the hash created by YAML parsing this file and also providing the path of this file in `new`, which returns a config. Then I call `validate` on the returned config.

## Expected behavior

I believe it should not throw an error since the hash passed is literally created from the config, or if its an error to do so, give a better error description.

## Actual behavior

It gives me a `unrecognized cop or department inherit_gem found in .rubocop.yml (RuboCop::ValidationError)`. However the error is not thrown when I call `validate_after_resolution`.

## Steps to reproduce the problem

```ruby
require 'rubocop'

conf = YAML.safe_load(File.read('.rubocop.yml'))
config = RuboCop::Config.new(conf, '.rubocop.yml')

config.validate

```

## RuboCop version

```
$ rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [x86_64-darwin21]
```

I'm not 100% sure if this is a bug, or if I'm using it wrong. I just couldn't find any documentation for this behaviour.

**Comment by NobodysNightmare** [comment]
Rubocop usually uses the `ConfigLoader` to create a configuration. This class also takes care of resolving configuration inheritance before initializing a `Config`.
